Bindings/Java/Matlab/Hopper_Device/RunHopperWithDevice_answers.m

```diff
@@ -36,11 +36,18 @@
 %% Connect the device to the hopper.
 % ----------------------------------
 
+% TODO: Add the device to the hopper model.
+% [Step 2, Task A]
+% ANSWER{
+hopper.addComponent(device);
```

Comment:
Note: I had to add the device to the hopper before connecting the sockets because, otherwise `connectSocket_*()` generates invalid relative paths. The invalid paths cause `findComponent()` to look up the component by name only, and since there are two components with the name `deviceAttachmentPoint`, `findComponent()` throws an exception.

As has been discussed elsewhere, `connectSocket()` should not attempt to generate a relative path; it should simply store a pointer to the connectee (or we should use UUIDs).

Reply:
I agree that we should only hang onto the reference when we connect. We should compute the path when it is time to serialize.

Reply:
> I agree that we should only hang onto the reference when we connect. We should compute the path when it is time to serialize.

To clarify, this is not for this PR, correct?

Reply:
Correct!